File: backend/microservice/src/admin/api/internal/RecordReviewAuditInternalApi.scala
package microservice.admin.api.internal

import cats.effect.IO
import java.sql.Connection
import microservice.admin.objects.submission.ReviewAudit
import microservice.admin.tables.AdminAuditTable
import microservice.infrastructure.api.{APIMessage, PlanSteps}
import microservice.infrastructure.http.HttpError

/** 模块间 API：写入审核审计；由 bird/level/admin HTTP API 调用，不挂路由。 */
final case class RecordReviewAuditInternalAPIMessage(
  targetType: String,
  submissionId: String,
  reviewerId: String,
  decision: String,
  reviewNote: Option[String],
  reviewedAt: String
) extends APIMessage[ReviewAudit] {
  override def plan(connection: Connection): IO[Either[HttpError, ReviewAudit]] =
    PlanSteps.finish {
      PlanSteps.read(
        ReviewAudit.from(
          AdminAuditTable.recordReview(
            connection = connection,
            targetType = targetType,
            submissionId = submissionId,
            reviewerId = reviewerId,
            decision = decision,
            reviewNote = reviewNote,
            reviewedAt = reviewedAt
          )
        )
      )
    }
}

File: frontend/src/api/admin/internal/RecordReviewAuditInternalApi.ts
import { z } from "zod";
import { APIMessage } from "../../../system/api/APIMessage.js";

/** Mirrors backend internal API admin/api/internal/RecordReviewAuditInternalApi.scala; not registered as a public frontend route. */
export class RecordReviewAuditInternalAPI extends APIMessage<unknown> {
  readonly targetType: unknown;
  readonly submissionId: unknown;
  readonly reviewerId: unknown;
  readonly decision: unknown;
  readonly reviewNote: unknown;
  readonly reviewedAt: unknown;
  constructor(targetType: unknown, submissionId: unknown, reviewerId: unknown, decision: unknown, reviewNote: unknown, reviewedAt: unknown) {
    super();
    this.targetType = targetType;
    this.submissionId = submissionId;
    this.reviewerId = reviewerId;
    this.decision = decision;
    this.reviewNote = reviewNote;
    this.reviewedAt = reviewedAt;
  }

  override get responseSchema() {
    return z.unknown();
  }
}
